WalletWasabi/Rpc/JsonRpcRequestHandler.cs

```diff
@@ -129,12 +134,18 @@ private async Task<string> HandleRequestAsync(JsonRpcRequest jsonRpcRequest, Can
 			{
 				if (!prodecureMetadata.MethodInfo.ReturnType.IsGenericType)
 				{
-					await ((Task)result).ConfigureAwait(false);
+					var task = result as Task
```

Comment:
We know that given the method is `async` is must return an instance of an "awaitable" thing (an type that implements `GetAwaitable` that returns something that implements `INotifyCompletion`). If well 99.9999% of the times we return `Task` it could be possible to return `ValueTask` or any other new awaitable and this will fail here.

Anyway, i don't think it worth it but we should not use reflection. Using reflection made the initial implementation easier but it became more and more complex with the time.